parts/linux/cloud-init/artifacts/cse_main.sh

```diff
@@ -215,11 +215,11 @@ if [[ "{{GetTargetEnvironment}}" == "AzureChinaCloud" ]]; then
     retagMCRImagesForChina
 fi
 
-{{- if EnableHostsConfigAgent}}
-logs_to_events "AKS.CSE.configPrivateClusterHosts" configPrivateClusterHosts
-{{- end}}
+if [[ "${ENABLE_HOSTS_CONFIG_AGENT}" == "true" ]]; then
+    logs_to_events "AKS.CSE.configPrivateClusterHosts" configPrivateClusterHosts
+fi
 
-{{- if ShouldConfigTransparentHugePage}}
+{{ if ShouldConfigTransparentHugePage -}}
```

Comment:
what ends up being the diff here?

Reply:
whitespace in generated files

the minus before the if leads to a missing newline between some bash
the minus after the if avoids removing the preceding newline, but trips everything inside the brackets